tests/jballerina-unit-test/src/test/resources/test-src/types/integer/integer-value-negative.bal

```diff
@@ -16,3 +16,27 @@ function test2() {
     // This is to verify that the compilation continues.
     int y = 1
 }
+
+type testType1 int:Signed32;
+type testType2 int:Signed16;
+type testType3 int:Signed8;
+
+function testStaticTypeOfUnaryExpr() {
+    int:Signed8 a = -128;
+    int:Signed8 _ = -a;
+    int:Signed16 b = -32768;
+    int:Signed16 _ = -b;
+    int:Signed32 c = -2147483648;
+    int:Signed32 _ = -c;
+
+    testType1 d = -2147483648;
+    int:Signed32 _ = -d;
+    testType2 e = -32768;
+    int:Signed16 _ = -e;
+    testType3 f = -128;
+    int:Signed8 _ = -f;
+
+    int:Signed16 _ = -(-32767);
+    int:Signed8 _ = -(-127);
+    int:Signed32 _ = -(-2147483647);
+}
```

Comment:
Let's comment where the error is (// error).

Reply:
The line from which the error is thrown is reflected in `BAssertUtil.validateError()` right, so do we need to again specify in the `bal` file?